src/frontend/src/optimizer/rule/rewrite_like_expr_rule.rs

```diff
@@ -0,0 +1,160 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::slice::memchr::memchr;
+use std::cmp::min;
+use std::str::from_utf8;
+
+use risingwave_common::types::ScalarImpl;
+use risingwave_connector::source::DataType;
+
+use super::{BoxedRule, Rule};
+use crate::expr::{Expr, ExprImpl, ExprRewriter, ExprType, ExprVisitor, FunctionCall, Literal};
+use crate::optimizer::plan_node::{ExprRewritable, LogicalFilter};
+use crate::optimizer::PlanRef;
+
+pub struct RewriteLikeExprRule {}
+impl Rule for RewriteLikeExprRule {
+    fn apply(&self, plan: PlanRef) -> Option<PlanRef> {
+        let filter: &LogicalFilter = plan.as_logical_filter()?;
+        let mut has_like = HasLikeExprVisitor {};
```

Comment:
Why do we visit first? Is it for avoiding clone?

Reply:
Yes, if the filter doesn't contains any like expression, just ignore it.